File: CS/SpreadsheetChartAPISamples/CodeUtils/FinalCall.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SpreadsheetChartAPISamples
{
    public static class FinalCallHelper
    {
        public static string CheckModule(ExampleLanguage lang, string regionName)
        {
            List<string> modules = new List<string> {""};
            if (modules.Contains(regionName))
            {
                if (lang == ExampleLanguage.Csharp) return finalCallCS;
                if (lang == ExampleLanguage.VB) return finalCallVB;
            }

            return string.Empty;
        }

            const string finalCallCS =
    "worksheet = workbook.Worksheets[\"blankSheet\"];\r\n" +
    "workbook.Worksheets.ActiveWorksheet = worksheet;\r\n" +
    "worksheet.Cells[\"B2\"].Value =" +
        "\"SpeadsheetControl does not visualize this sample correctly.\" + Environment.NewLine +" +
        "\"However, property values are loaded and stored in supported formats,\" + Environment.NewLine +" +
        "\"and you can modify them programmatically.\";";

            const string finalCallVB =
    "worksheet = workbook.Worksheets(\"blankSheet\")\r\n" +
    "workbook.Worksheets.ActiveWorksheet = worksheet\r\n" +
    "worksheet.Cells(\"B2\").Value = " +
    "\"SpeadsheetControl does not visualize this sample correctly.\"" +
    "& Constants.vbCrLf & \"However, property values are loaded and stored in supported formats,\"" +
    "& Constants.vbCrLf & \"and you can modify them programmatically.\"";
        }

}
    

File: VB/SpreadsheetChartAPISamples/CodeUtils/FinalCall.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace SpreadsheetChartAPISamples
	Public NotInheritable Class FinalCallHelper
		Private Sub New()
		End Sub
		Public Shared Function CheckModule(ByVal lang As ExampleLanguage, ByVal regionName As String) As String
            Dim modules As New List(Of String)(New String() {""})
			If modules.Contains(regionName) Then
				If lang = ExampleLanguage.Csharp Then
					Return finalCallCS
				End If
				If lang = ExampleLanguage.VB Then
					Return finalCallVB
				End If
			End If

			Return String.Empty
		End Function

        Private Const finalCallCS As String = "worksheet = workbook.Worksheets[""blankSheet""];" & Constants.vbCrLf & "workbook.Worksheets.ActiveWorksheet = worksheet;" & Constants.vbCrLf & "worksheet.Cells[""B2""].Value =" & """SpeadsheetControl does not visualize this sample correctly."" + Environment.NewLine +" & """However, property values are loaded and stored in supported formats,"" + Environment.NewLine +" & """and you can modify them programmatically."";"

        Private Const finalCallVB As String = "worksheet = workbook.Worksheets(""blankSheet"")" & Constants.vbCrLf & "workbook.Worksheets.ActiveWorksheet = worksheet" & Constants.vbCrLf & "worksheet.Cells(""B2"").Value = " & """SpeadsheetControl does not visualize this sample correctly.""" & "& Constants.vbCrLf & ""However, property values are loaded and stored in supported formats,""" & "& Constants.vbCrLf & ""and you can modify them programmatically."""
	End Class

End Namespace